File: LC_JS-SOLUTIONS/929. Unique Email Addresses.js
/**
 * @param {string[]} emails
 * @return {number}
 */
var numUniqueEmails = function (emails) {
  let eSet = new Set();

  for (let email of emails) {
    let [local, domain] = email.split("@");

    let pIndex = local.indexOf("+");
    if (pIndex !== -1) {
      local = local.substring(0, pIndex);
    }

    local = local.replace(/\./g, "");
    final = local + "@" + domain;
    eSet.add(final);
  }
  return eSet.size;
};

File: LC-Python-Solution/929. Unique Email Addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:

        eSet = set()

        for email in emails:
            name,domain = email.split("@")

            atIndex = name.find("+")
            if atIndex !=-1:
                name = name[:atIndex]
            
            name = name.replace(".","")
            eSet.add(name+"@"+domain)
        
        return len(eSet)




        